python/ray/data/read_api.py

```diff
@@ -48,26 +60,14 @@
 from ray.data.dataset import Dataset, MaterializedDataset
 from ray.data.datasource import (
     BaseFileMetadataProvider,
-    BigQueryDatasource,
-    BinaryDatasource,
     Connection,
-    CSVDatasource,
     Datasource,
     ImageDatasource,
```

Comment:
`ImageDatasource`, `LancecDatasource`, `TFRecordDatasource`, and `TorchDatasource` are annotated as developer APIs but they should be internal. Could we remove the annotations and also move them to `_internal`? (Saw there's already a PR for `TFRecordDatasource`: https://github.com/ray-project/ray/pull/46171)

Reply:
absolutely

Reply:
actually if it's ok let me do it in a follow up PR to reduce the size of this one

Reply:
https://github.com/ray-project/ray/pull/46328/files